meson.build

```diff
@@ -180,6 +180,9 @@ endif
 
 if get_option('enable-uefi')
   fwup = dependency('fwup', version : '>= 5')
+  if fwup.version().version_compare('>= 9')
+    conf.set('HAVE_FWUPD_GET_BGRT_INFO', '1')
```

Comment:
Also, this should be `HAVE_FWUP_GET_BGRT_INFO`.

Reply:
Good catch about the name. The new functionality is only in the ux-capsule branch of fwupdate, but Peter wanted me to see if it was suitable ahead of merging to master. I assume this probably needs to be `10` when it hits master.

Reply:
@vathpela can you bump the version in master please and I'll adjust this patch to be 10 rather than 9.

Reply:
Huh, why didn't CI fail then?  `fwup_get_ux_capsule_info` shouldn't have been in 9 which should have caused a UEFI compilation failure.

Reply:
Good question!

Reply:
The builders only have `Native dependency fwup found: YES 8` so we're using the fallback 1024x768 code.

Reply:
That doesn't seem right to me still.  fwupdate 9 is in debian.  At least the Debian one should have failed then.
See: https://tracker.debian.org/pkg/fwupdate

Reply:
You're correct `Native dependency fwup found: YES 9` on https://travis-ci.org/hughsie/fwupd/jobs/272904150 (debian):

    ../plugins/uefi/fu-plugin-uefi.c:287:7: warning: implicit declaration of function ‘fwup_get_ux_capsule_info’; did you mean ‘fwup_print_update_info’? [-Wimplicit-function-declaration]
      rc = fwup_get_ux_capsule_info (&screen_width, &screen_height);
           ^~~~~~~~~~~~~~~~~~~~~~~~
           fwup_print_update_info
    ../plugins/uefi/fu-plugin-uefi.c:287:7: warning: nested extern declaration of ‘fwup_get_ux_capsule_info’ [-Wnested-externs]

That probably needs to become an error somehow.

Reply:
Or maybe just having -Werror on for CI builds? That might be too aggressive and throw up too many failures with very old / very new gcc versions. Talking of which, one of the CI targets should probably be using clang rather than GCC too.

Reply:
So, -Werror didn't work out so well, Debian has https://travis-ci.org/hughsie/fwupd/jobs/272942674:

    /usr/include/smbios_c/config/compiler/gcc.h:31:7: error: #warning "Unknown compiler version - please run the configure tests and report the results" [-Werror=cpp]

and Arch has https://travis-ci.org/hughsie/fwupd/jobs/272942675:

    /usr/include/features.h:373:4: error: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Werror=cpp]

Reply:
Yeah that libsmbios thing is dumb.  It's been removed from master, but a new release hasn't been done in a while.  I think some of @vathpela's fixes actually caused some problems too so it's probably better not to do a new release until those are resolved.

I think ideally a 4th build target should be added for these various deviations.  The first 3 are stuff that will actually be packaged in the distros that participate.  The 4th target can just be a few rapid test compiles with clang, with more error checking turned on, with different dep versions etc.

Reply:
I think https://github.com/hughsie/fwupd/pull/228 does the right thing, i.e. just enables all warnings as errors, except cpp errors.